src/start_server.js

```diff
@@ -13,24 +13,54 @@ var http = require('http');
 var makeApp = require('./make_app');
 var open = require('open');
 var util = require('util');
+var findPort = require('find-port');
 
-function startServer(port, openPage, componentDir, packageName) {
-  port = port || 8080;
-  console.log('Starting Polyserve on port ' + port);
+function startWithPort(options) {
+
+  options.port = options.port || 8080;
+
+  console.log('Starting Polyserve on port ' + options.port);
 
   var app = express();
-  var polyserve = makeApp(componentDir, packageName);
+  var polyserve = makeApp(options.componentDir, options.packageName);
 
   app.use('/components/', polyserve);
 
   var server = http.createServer(app);
-  server = app.listen(port);
 
-  var baseUrl = util.format('http://localhost:%d/components/%s/', port, polyserve.packageName);
+  server = app.listen(options.port);
+
+  server.on('error', function(err) {
+    if (err.code === 'EADDRINUSE')
+      console.error("ERROR: Port in use", options.port, "Aborting!");
```

Comment:
Let's replace "Aborting!" with something like "Please choose another port, let an unused port be chosen automatically"